GameDesign/Assets/Scripts/MonoBehaviours/CustomerMonoBehavior.cs

```diff
@@ -11,21 +11,45 @@ public class CustomerMonoBehavior : MonoBehaviour, Clickable
     [SerializeField] public int requiredClicks = 1;
     [SerializeField] public int clickTime;
     [SerializeField] public bool wearsMask;
-    [SerializeField] public NPCMovementManager moveManager;
     [SerializeField] public int pointValue;
     [SerializeField] public bool isFrozen = false;
     [SerializeField] public int frozenTimeCount = 0;
 
     [SerializeField] public TaserManager taserManager = TaserManager.Instance;
 
+    [SerializeField] private FiniteStateMachine<CustomerMonoBehavior> fsm;
+
     void Start()
     {
         InvokeRepeating("UnFreeze", 1.0f, 1.0f);
```

Comment:
the invoke repeating doesnt work as a cooldown tracker. its based on a clock from creation of the object, we should have a separate counter from when it gets hit for x time where it will stay frozen.
because we are not sure in what moment of the cycle the character will be hit. if its at the start in will be about 1 second but if its at the end it will be close to 0.

Reply:
I don't see the problem nor see an alternative to using InvokeRepeating

Reply:
alternative use a simple coroutine that that is called when you want to freeze the character and after the time it unfreezes it.
alla IEnumerator coroutineName(){
  freeze()
 yield return new waitForSeconds(x)
  unfreeze()
}
"problem" InvokeRepeating is called (as far as this code shows) every second. 
so the level starts at 0:00:00 (m:ss:msms)  and every second it will work. 
problem if we freeze a character for example at x:10:90 the InvokeRepeating will be called at x:11:00 so the "freeze" will last only 0.1 seconds, which is hard to perceive, and in theory is not even the most extreme case.